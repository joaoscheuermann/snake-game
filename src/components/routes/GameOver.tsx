import { Button, Flex, Heading } from "@radix-ui/themes";
import { GameStatus, useGameStateStore } from "../../stores/state";
import { useNavigate } from "react-router";

export default function GameOver() {
  const navigate = useNavigate();
  const game = useGameStateStore();

  const score = useGameStateStore((state) => state.score);
  const status = useGameStateStore((state) => state.status);

  const messages: Record<GameStatus, string | null> = {
    [GameStatus.PlayerWon]: "Congratulations, you won!",
    [GameStatus.PlayerLost]: "You lost, try again!",
    [GameStatus.Stoped]: null,
    [GameStatus.Paused]: null,
    [GameStatus.Running]: null,
  };

  const message = messages[status];

  function handlePlayAgainButton() {
    game.stop();
    navigate("/play");
  }

  return (
    <Flex
      direction="column"
      align="center"
      justify="center"
      className="h-screen"
      gap="6"
    >
      <Heading size="9" className="mb-4">
        {message}
      </Heading>
      <Heading size="4" className="mb-8">
        Your score: {score}
      </Heading>
      <Button size="3" variant="surface" onClick={handlePlayAgainButton}>
        Play again!
      </Button>
    </Flex>
  );
}
