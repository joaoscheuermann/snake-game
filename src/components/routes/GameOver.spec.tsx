import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import GameOver from "./GameOver";

describe("GameOver", () => {
  it("should render correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <GameOver />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
