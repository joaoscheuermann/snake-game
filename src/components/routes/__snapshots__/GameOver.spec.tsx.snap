// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOver should render correctly 1`] = `
<DocumentFragment>
  <div
    class="rt-Flex rt-r-fd-column rt-r-ai-center rt-r-jc-center rt-r-gap-6 h-screen"
  >
    <h1
      class="rt-Heading rt-r-size-9 mb-4"
    />
    <h1
      class="rt-Heading rt-r-size-4 mb-8"
    >
      Your score: 
    </h1>
    <button
      class="rt-reset rt-BaseButton rt-r-size-3 rt-variant-surface rt-Button"
      data-accent-color=""
    >
      Play again!
    </button>
  </div>
</DocumentFragment>
`;
